### PR TITLE
8269131: Update libxml2 to version 2.9.12

### PR DESCRIPTION
**Pull request by arapte**
We currently use libxml2 version 2.9.10. It should be updated to latest stable release, which is version 2.9.12.
The steps to update libxml are documented in UPDATING.txt.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269131](https://bugs.openjdk.java.net/browse/JDK-8269131): Update libxml2 to version 2.9.12

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/562/head:pull/562` \
`$ git checkout pull/562`

Update a local copy of the PR: \
`$ git checkout pull/562` \
`$ git pull https://git.openjdk.java.net/jfx pull/562/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 562`

View PR using the GUI difftool: \
`$ git pr show -t 562`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/562.diff">https://git.openjdk.java.net/jfx/pull/562.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Updating libxml in OpenJFX:
+
+1. Download respective libxml source tarball from here: http://xmlsoft.org/sources
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt:3]: As an alternative, the source can be downloaded by cloning:
https://gitlab.gnome.org/GNOME/libxml2.git
They tag each release. Maybe it isn't worth noting here.

**Review by kevinrushforth — COMMENTED** [review]
The changes look good except for the expansion of tags in `Makefile.am` (we don't want to expand tabs in Makefiles). I recommend re-copying `Makefile.am` from the original sources of libxml 2.9.12. I left a couple questions on the new instructions in `UPDATING.txt` as well.

I did a build / test on all 3 platforms. No issues observed.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/src/Makefile.am [hunk 2]
```diff
@@ -46,16 +46,16 @@ trio_sources =
...
-		valid.c xlink.c HTMLparser.c HTMLtree.c debugXML.c xpath.c  \
-		xpointer.c xinclude.c nanohttp.c nanoftp.c \
-		$(docb_sources) \
-		catalog.c globals.c threads.c c14n.c xmlstring.c buf.c \
-		xmlregexp.c xmlschemas.c xmlschemastypes.c xmlunicode.c \
-		$(trio_sources) \
-		xmlreader.c relaxng.c dict.c SAX2.c \
-		xmlwriter.c legacy.c chvalid.c pattern.c xmlsave.c \
-		xmlmodule.c schematron.c xzlib.c
+        parser.c tree.c hash.c list.c xmlIO.c xmlmemory.c uri.c  \
```
[the remark this answers is not in the document]
Comment (arapte) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/src/Makefile.am]: I think I ran a wrong command to cause this, It is corrected now.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+6.2 Copy libxml\src\config.h to libxml\linux\config.h
+
+7. Helper commands for removing leading tabs and trailing whitespaces.
+7.1 > cd modules/javafx.web/src/main/native/Source/ThirdParty/libxml
+7.2 Remove tabs from source files:
+    > sudo apt install moreutils
+    > find src/ -name "*.c" -type f -exec bash -c 'expand -t 4 "$0" | sponge "$0"' {} \;
+    > find src/ -name "*.h" -type f -exec bash -c 'expand -t 4 "$0" | sponge "$0"' {} \;
+7.3 Remove trailing whitespaces:
+    > find . -type f -exec sed -i '' 's/[[:space:]]\{1,\}$//' {} \+
```
[the remark this answers is not in the document]
Comment (arapte) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt]: `Makefile.am` tab expansion was a result of running a wrong command, but not this. Also I changed this command to remove trailing whitespaces only from source files.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        > cd modules/javafx.web/src/main/native/Source/ThirdParty/libxml/src/win32
+        > cscript configure.js compiler=msvc catalog=no xml_debug=no docb=no ftp=no http=no iconv=no legacy=no mem_debug=no modules=no regexps=no run_debug=no schemas=no schematron=no valid=no xinclude=no xptr=no zlib=no
+
+        - Above command generates a header file `libxml\src\include\libxml\xmlversion.h` and may update the `src/config.h` file. (on all platforms)
+4.1 Copy `libxml\src\include\libxml\xmlversion.h` to `libxml\win32\include\libxml\xmlversion.h'. xmlversion.h file defines several macros to control libxml features. We do not require all of the features to be enabled. Compare the generated `libxml\src\include\libxml\xmlversion.h` with existing `libxml\win32\include\libxml\xmlversion.h' and retain macro definitions from the later.
+
+4.2 Copy `libxml\src\config.h` to `libxml\win32\config.h`
+
+5. Remove files & directories which are not relevant to JavaFX WebKit.
+   Use git clean command. (git clean -df modules/)
```
[the remark this answers is not in the document]
Comment (arapte) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt]: I removed the use of git clean command from the steps.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt (hunk 1 above)
Comment (arapte) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt]: Added the git url also as an alternate source in the steps.

**Review by kevinrushforth — APPROVED** [review]

**Review by johanvos — APPROVED** [review]
confirmed to work on linux, win, mac (x86_64)

**Pull request closed without merging**